src/plugins/intel_gna/src/request/worker_impl.cpp

```diff
@@ -42,23 +43,36 @@ Gna2Model* WorkerImpl::model() {
 void WorkerImpl::enqueueRequest() {
     check_if_free();
 
-    for (auto& subrequest : modelSubrequests_) {
-        subrequest->enqueue();
+    try {
+        for (auto& subrequest : modelSubrequests_) {
+            subrequest->enqueue();
+        }
+    } catch (const std::exception& e) {
+        ov::intel_gna::log::error() << "Exception when enqueueing inference: " << e.what() << std::endl;
+        cleanup_unfinished_submodules();
+        throw;
```

Comment:
Do we have test for that?

Reply:
Reworked worker. 
Tests for it are done unit tests gna_request_worker_impl_test.cpp